File: samples/go/04primality/prime.go
package main

import (
    "fmt"
    "math"
    "os"
    "strconv"
)


func isPrime(number float64) (bool) {
    var limit float64 = math.Sqrt(number);
    for i := 2.0; i <= limit; i++ {
        if math.Mod(number, i) == 0 {
            return false;
        }
    }
    return true;
}


func main() {
    if len(os.Args) < 2 {
        fmt.Println("Error: Usage ./prime <number> [<number> ...]");
        os.Exit(1);
    }

    for _, arg := range os.Args[1:] {
        number, err := strconv.ParseFloat(arg, 64)
        if err != nil {
            fmt.Printf("Error: Expected to receive a number, recevied '%s'\n", arg);
            os.Exit(2);
        }

        var primestring = "false";
        if isPrime(number) {
            primestring = "true";
        }

        fmt.Printf("%f => %s\n", number, primestring);
    }
}

File: samples/rust/04primality/prime.rs
use std::env;

fn is_prime(num: i32) -> bool {
    let limit: i32 = (num as f64).sqrt() as i32;
    for n in 2 .. limit + 1 {
        if num % n == 0 {
            return false;
        }
    }
    return true;
}

fn main() {
    let args: Vec<String> = env::args().collect();

    if args.len() == 1 {
        eprintln!("Error: Usage {} <number> [<number> ...]", args[0]);
        std::process::exit(1);
    }

    for arg in &args[1 .. args.len()] {
        let result = arg.parse::<i32>();
        match result {
            Ok(v) if v > 1 => println!("{} => {}", v, if is_prime(v) { "true" } else { "false" }),
            _ => {
                eprintln!("Error: Expected to receive an integer greater than 1. Received {}", arg);
                std::process::exit(2);
            }
        }
    }
}
